### PR TITLE
Support @deprecated tag in java wrappers

### LINKED ISSUE #9699: Deprecated interfaces not marked as deprecated in java wrappers
##### System information (version)

- OpenCV => 3.3
- Operating System / Platform => all
- Compiler => all

##### Detailed description

Currently java wrapper generator ignores doxygen tags (`\deprecated` or `@deprecated`). Desired behaviour - any class or interface deprecated in OpenCV API shall be also tagged as `@Deprecated` in the wrapper.

##### Steps to reproduce
Build OpenCV from sources with java enabled. Any deprecated class (for example cv::dnn::Importer) shall not be marked as `@Deprecated` in the corresponding java wrapper.

### PR DESCRIPTION
**Pull request by abratchik**
resolves #9699 

### This pullrequest changes

gen_java.py

Deprecated interfaces in C/C++ code  shall be declared with @Deprecated tag in java wrappers thus encouraging java community to utilize new API.

Tests: FeatureDetector and DescriptorExtractor in features2d_maunual.hpp deprecated as no longer required - direct istantiation of Feature2D covers all the cases and is more flexible.

### CODE REVIEW

**Comment by vpisarev** [comment]
@mshabunin, could you please review it?

**Comment by abratchik** [comment]
Gentlemen, would you merge this please? It is really minor change but will help to manage Java API properly. Or alternatively let me know if anything is incorrect.

File: modules/java/generator/gen_java.py [hunk 1]
```diff
@@ -182,8 +181,20 @@ def libVersionBlock():
...
+
+        # parse doxygen comments
+        self.params={}
+        if type == "class":
+            docstring="// C++: class " + self.name + "\n//javadoc: " + self.name
+        else:
+            docstring=""
+        if len(decl)>5 and decl[5]:
+            if re.search("(@|\\\\)deprecated", decl[5]):
+                docstring += "\n@Deprecated"
```
Comment (mshabunin) [inline modules/java/generator/gen_java.py]: It is better to create additional `annotation` field instead of putting it to the `docstring`.

**Comment by abratchik** [comment]
It looks like an issue on Linux OpenCL environment - it is failing tests across all pull requests.

**Pull request merged**